src/app/cli/src/init/coda_run.ml

```diff
@@ -115,13 +122,68 @@ let get_current_fork_id ~compile_time_current_fork_id ~conf_dir ~logger =
            config; please delete your Coda config if you wish to use a new \
            fork ID" )
     with Sys_error _ ->
-      (* use value provided on command line, write to config dir, possibly overwriting existing entry *)
+      (* use value provided on command line, write to config dir *)
+      if Option.is_none (Fork_id.create_opt fork_id) then (
+        Logger.fatal logger ~module_:__MODULE__ ~location:__LOC__
+          "Fork ID provided on command line is invalid"
+          ~metadata:[("fork_id", `String fork_id)] ;
+        failwith "Fork ID from command line is invalid" ) ;
       write_fork_id fork_id ;
       Logger.info logger ~module_:__MODULE__ ~location:__LOC__
         "Using current fork ID $fork_id from command line, writing to config"
         ~metadata:[("fork_id", `String fork_id)] ;
       fork_id )
 
+let get_next_fork_id_opt ~conf_dir ~logger =
+  let read_fork_id, write_fork_id =
+    make_conf_dir_item_io ~conf_dir ~filename:"next_fork_id"
+  in
+  function
+  | None -> (
+    try
+      (* not provided on command line, try to read from config dir *)
+      let fork_id = read_fork_id () in
+      Logger.info logger ~module_:__MODULE__ ~location:__LOC__
+        "Setting next fork ID to $fork_id from config"
+        ~metadata:[("fork_id", `String fork_id)] ;
+      Some fork_id
+    with Sys_error _ ->
+      (* not on command-line, not in config dir, there's no next fork ID *)
+      None )
+  | Some fork_id -> (
+    try
+      (* it's an error if the command line value disagrees with the value in the config *)
+      let config_fork_id = read_fork_id () in
+      if String.equal config_fork_id fork_id then (
+        Logger.info logger ~module_:__MODULE__ ~location:__LOC__
+          "Using next fork ID $fork_id from command line, which matches the \
+           one in the config"
+          ~metadata:[("fork_id", `String fork_id)] ;
+        Some config_fork_id )
+      else (
+        Logger.fatal logger ~module_:__MODULE__ ~location:__LOC__
+          "Next fork ID $fork_id from the command line disagrees with \
+           $config_fork_id from the Coda config"
+          ~metadata:
+            [ ("fork_id", `String fork_id)
+            ; ("config_fork_id", `String config_fork_id) ] ;
+        failwith
```

Comment:
We should add a force flag which will overwrite the existing next fork id in the config. Otherwise, once someone sets what fork to vote for, they will not be able to change it, which seems undesirable. Debatably, the behavior here should be to just overwrite the config instead of crashing (which would mean you wouldn't need a force flag).

Reply:
@nholland94 in the interest of minimizing flags / cognitive load, I went with the overwrite idea.